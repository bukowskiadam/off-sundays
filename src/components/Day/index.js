import React from 'react';
import PropTypes from 'prop-types';

import StatusImage from '../StatusImage';
import { formatForHuman } from '../../helpers/formatDate';
import style from './index.module.css';

const DateHeader = ({ date }) => {
    const { day: day, date: formattedDate } = formatForHuman(date);
    return (
        <div>
            <h2 className={style.day}>{day}</h2>
            <h4 className={style.date}>{formattedDate}</h4>
        </div>
    );
};

const Status = ({ isClosed }) => (
    <div>
        <div className={style.image}>
            <StatusImage isClosed={isClosed} />
        </div>
        <div>{isClosed ? 'Handel ograniczony' : 'Handel dozwolony'}</div>
    </div>
);

const Day = ({ date, isClosed }) => (
    <div className={style.container}>
        <DateHeader date={date} />
        <Status isClosed={isClosed} />
    </div>
);

Day.propTypes = {
    date: PropTypes.instanceOf(Date).isRequired,
    isClosed: PropTypes.bool.isRequired,
};

export default Day;
