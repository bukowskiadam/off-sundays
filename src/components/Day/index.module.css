.container {
    padding: 20px 10px;
}

.day {
    margin-bottom: 0.2em;
}

.date {
    margin-bottom: 0.8em;
}

.image {
    max-width: 150px;
    margin: auto;
}

.image img {
    margin-bottom: 0;
}
