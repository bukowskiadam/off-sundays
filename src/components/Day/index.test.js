import React from 'react';
import { render } from 'enzyme';

import Day from './index';

describe('Day', () => {
    const date = new Date('2018-03-08T12:00:00.000Z');

    it('renders correctly when shops are closed', () => {
        const wrapper = render(<Day date={date} isClosed={true} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly when shops are opened', () => {
        const wrapper = render(<Day date={date} isClosed={false} />);

        expect(wrapper).toMatchSnapshot();
    });
});
