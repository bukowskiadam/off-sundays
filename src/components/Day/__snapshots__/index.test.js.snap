// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day renders correctly when shops are closed 1`] = `
<div
  class="container"
>
  <div>
    <h2
      class="day"
    >
      Czwartek
    </h2>
    <h4
      class="date"
    >
      8 marca 2018 r.
    </h4>
  </div>
  <div>
    <div
      class="image"
    >
      <img
        alt="zamknięte"
        src="icon-closed.svg"
      />
    </div>
    <div>
      Handel ograniczony
    </div>
  </div>
</div>
`;

exports[`Day renders correctly when shops are opened 1`] = `
<div
  class="container"
>
  <div>
    <h2
      class="day"
    >
      Czwartek
    </h2>
    <h4
      class="date"
    >
      8 marca 2018 r.
    </h4>
  </div>
  <div>
    <div
      class="image"
    >
      <img
        alt="otwarte"
        src="icon-opened.svg"
      />
    </div>
    <div>
      Handel dozwolony
    </div>
  </div>
</div>
`;
